### PR TITLE
Fix AssertionError in evict Tests (resolves #925)

### LINKED ISSUE #925: AssertionError in GoogleJobStoreCacheTest
http://jenkins.cgcloud.info/job/toil-pull-requests/1389/testReport/junit/src.toil.test.src.jobCacheTest/GoogleJobStoreCacheTest/testCacheEvictionFailCase/

### PR DESCRIPTION
**Pull request by arkal**
resolves #925

Modified the test to have no retries, leading to only a single failed job instead of 2.
Modified the delete mutably and immutably read file tests to not be non-deterministic

### CODE REVIEW

File: src/toil/test/src/jobCacheTest.py [hunk 1]
```diff
@@ -731,15 +727,18 @@ def _removeReadFileFn(job, fileToDelete, readAsMutable):
...
             # behaviour is seen in both conditions.
             while True:
-                os.remove(outfile)
+                os.rename(outfile, tempfile)
                 try:
                     job.fileStore.deleteLocalFile(fileToDelete)
                 except IllegalDeletionCacheError:
+                    job.fileStore.logToMaster('Detected a deleted file %s.' % fileToDelete)
+                    os.rename(tempfile, outfile)
                     pass
```
Comment (hannes-ucsc) [inline src/toil/test/src/jobCacheTest.py]: `pass` is redundant

File: src/toil/test/src/jobCacheTest.py [hunk 2]
```diff
@@ -684,8 +685,8 @@ def _controlledFailTestFn(job, jobDisk, testDir):
                 Hidden.AbstractCacheTest._requirementsConcur(job, jobDisk, cached)
                 Hidden.AbstractCacheTest._returnFileTestFn(job, jobDisk, cached, testDir, 20)
             else:
-                modifiedJobReqs, cached = Hidden.AbstractCacheTest._returnFileTestFn(job, jobDisk, 0,
-                                                                                     testDir, 20)
+                modifiedJobReqs, cached = Hidden.AbstractCacheTest._returnFileTestFn(job, jobDisk,
```
Comment (hannes-ucsc) [inline src/toil/test/src/jobCacheTest.py]: Might want to extract `Hidden.AbstractCacheTest` into a variable so you don't have to wrap the arguments so narrowly.

**Comment by hannes-ucsc** [comment]
Test error tracked in #933 and #934.

@arkal can you explain why the word `tests` is uppercase in your commit message? Please fix. Also, it should be "eviction" not "evict".

**Comment by hannes-ucsc** [comment]
"cache eviction" actually.

**Comment by arkal** [comment]
@hannes-ucsc i implemented the suggested changes

**Comment by hannes-ucsc** [comment]
Build failures: #934 and #579

**Pull request merged**